pkg/listen/listen.go

```diff
@@ -17,77 +17,111 @@ package listen
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"net/url"
 	"regexp"
+	"strings"
 
-	"github.com/hookdeck/hookdeck-cli/pkg/ansi"
 	"github.com/hookdeck/hookdeck-cli/pkg/config"
 	"github.com/hookdeck/hookdeck-cli/pkg/login"
 	"github.com/hookdeck/hookdeck-cli/pkg/proxy"
+	hookdecksdk "github.com/hookdeck/hookdeck-go-sdk"
 	log "github.com/sirupsen/logrus"
 )
 
 type Flags struct {
-	NoWSS bool
+	NoWSS   bool
+	CliPath string
 }
 
 // listenCmd represents the listen command
-func Listen(URL *url.URL, source_alias string, connectionQuery string, flags Flags, config *config.Config) error {
+func Listen(URL *url.URL, sourceQuery string, connectionFilterString string, flags Flags, config *config.Config) error {
 	var err error
-	var guest_url string
+	var guestURL string
+
+	sourceAliases, err := parseSourceQuery(sourceQuery)
+	if err != nil {
+		return err
+	}
+
+	isMultiSource := len(sourceAliases) > 1 || (len(sourceAliases) == 1 && sourceAliases[0] == "*")
+
+	if flags.CliPath != "" {
+		if isMultiSource {
+			return errors.New("Can only set a CLI path when listening to a single source")
+		}
+
+		_, err = isPath(flags.CliPath)
+		if err != nil {
+			return errors.New("The CLI path must be in a valid format")
+		}
```

Comment:
![CleanShot 2024-07-29 at 19 52 57](https://github.com/user-attachments/assets/6e6b619a-dd50-471d-9ca3-ad8866193b76)

Just confirming if the path without `/` is valid. I think so but it does seem a bit awkward.

Reply:
Got it, yeah that's fair!